### DISCUSSION TITLE
Conversational AI assistant

### DISCUSSION THREAD

**Discussion by saltcod** [supabase/supabase#21967] [General]
In Launch Week 7, we added Supabase AI to the SQL Editor. It was a brand new way to write SQL (by not writing it all). This is one of our most-used tools, and we get a lot of feedback about it. We’ve spent some time working on an iteration over the past few weeks and we’d love to know what you think.

**What we heard:**
- the one-shot prompt is fine, but a conversational ui would be better
- the one-shot prompt slowed down workflow (again, conversation = better)
- it wasn’t obvious whether or not you were sending your project metadata 

**What we'd like to know:**
- does the new conversational Assistant help? 
- does it feel much faster now? Does it keep you in the flow?
- how are the AI responses? Are you seeing areas that could be improved?

**Try it out!**
You can enable the new conversational Assistant from the Feature Preview:
 
![311246985-2ed278a5-7d51-4676-946b-c522776d5a32](https://github.com/supabase/supabase/assets/105593/cab27bf6-365b-4b52-b876-aca57d72b332)
![screenshot-2024-03-12-at-10 02 38](https://github.com/supabase/supabase/assets/105593/32c81341-597f-4caf-8a2c-8b03bebbcb80)

**Comment by madebyfabian** [comment]
Just used it, very cool! One thing though, I cannot copy anything out of the right window. Whether its part of the code or any message I wrote for example. When selecting the text and hitting CMD + C, it does not get copied.

**Reply by madebyfabian** [reply]
General feedback: It definitely helps to get a better overview about where we came from and how the answers evolved. However, since it's never returning anything besides the SQL, sometimes it's hard to know if it really understood me well

**Reply by Hallidayo** [reply]
Hi @madebyfabian - Just checking does the copying issue still happen for you? I am able to my end, also if you hover over the SQL generated by the AI their should be a copy icon that you can use.

**Reply by madebyfabian** [reply]
@Hallidayo Hi there, I don't have it anymore, just checked. Works fine now :)

**Comment by rajivsinclair** [comment]
Is there going to be any way to bring this feature (and other AI-powered magic) to a self-hosted instance of Supabase? Of course, I am assuming that if the underlying LLM is not local/open-source, then we would need to configure our own API keys from OpenAI/Anthropic/Cohere/etc.

**Reply by rosmur** [reply]
and is it just querying GPT-4 under the hood? Or any specialized model - there are a few fine tuned models claiming great NL2SQL performance

**Reply by rajivsinclair** [reply]
Looks to me like it uses the GPT-3.5 model from OpenAI: https://github.com/supabase/supabase/blob/master/packages/ai-commands/src/sql.ts

**Comment by rajivsinclair** [comment]
For a self-hosted instance, can I just specify my API key in the .env for the docker containers that Supabase spins up?

**Reply by aweeden** [reply]
+1

**Reply by lazmo88** [reply]
Following, is there no docs available for these new AI features? Or are these not available for the self hosted version?

**Reply by volfadar** [reply]
For those of you searching for the environment variable to include in the docker-compose.yml file, you should add `OPENAI_API_KEY` under the `supabase-studio` service section
https://github.com/supabase/supabase/blob/f6d96174135fddc2c2bf8a7dd59225a02aee5559/docker/docker-compose.yml#L34
